rule.php

```diff
@@ -277,6 +272,13 @@ public function prevent_access() {
             return false;
         }
 
+        // If global setting to require a quiz password is set, check that it is set at global level or quiz level.
```

Comment:
Seems like it's a wring place to check that setting. User story looks like:


[SA7] As an admin, I want to be able to enforce that a quiz must have a password set if it is a QRSEB, so that I can prevent the teacher from forgetting to set a quiz password. | Activate the admin setting. Then, as teacher, try to set up a QRSEB without defining a quiz password. The user interface gives feedback and isn’t allowing saving the quiz settings.
-- | --

We need to move it a quiz form.

Reply:
The current version pushed now performs form validation if the global setting for quiz passwords is enabled.